src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1299,7 +1299,18 @@ protected internal override void OnProjectRemoved(ProjectId projectId)
         {
             lock (_gate)
             {
-                _projectReferenceInfo.Remove(projectId);
+                if (_projectReferenceInfo.TryGetValue(projectId, out var projectReferenceInfo))
```

Comment:
Can we use a better name for the local here? Having both `_projectReferenceInfo` and `projectReferenceInfo` is subtle.

Reply:
Fixed.

Reply:
(by renaming the outer map)

Reply:
Thanks!